cmd/appliance/upgrade/prepare.go

```diff
@@ -208,9 +207,12 @@ func prepareRun(cmd *cobra.Command, args []string, opts *prepareUpgradeOptions)
 	if err != nil {
 		return err
 	}
+	if len(opts.actualHostname) > 0 {
+		host = opts.actualHostname
+	}
```

Comment:
This doesn't override the host actually used to establish the connection Right?
This should just override the host when running `FindPrimaryController` and for the `controller://` file download URL.

Reply:
No, it's only used for finding the primary controller and then never used again.

Reply:
OK we just need to make sure both approaches come up with the same answer.

Reply:
the other one depends on the former, so yes